src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -233,6 +224,51 @@ public async Task UpdateSearchParameterIndicesBatchAsync(IReadOnlyCollection<Res
             }
         }
 
+        public async Task<ResourceWrapper> UpdateSearchParameterForResourceAsync(ResourceWrapper resourceWrapper, WeakETag weakETag, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(resourceWrapper, nameof(resourceWrapper));
+            EnsureArg.IsNotNull(weakETag, nameof(weakETag));
+
+            var cosmosWrapper = new FhirCosmosResourceWrapper(resourceWrapper);
+            ItemRequestOptions requestOptions = new ItemRequestOptions()
+            {
+                IfMatchEtag = weakETag.VersionId,
```

Comment:
I don't think this will work, `IfMatchEtag` refers to the `_etag` property of the Cosmos document

Reply:
It works currently (I have a test also that specifically tests this scenario - `GivenAnUpdatedResourceWithWrongETag_WhenUpdateSearchParameterForResourceAsync_ThenExceptionIsThrown` ). It basically depends on the caller creating the `WeakETag` with the appropriate result from the CosmosDb document. But I get your point that using the `WeakETag` class here does seem confusing (since that class is supposed to be a wrapper for version id). I have updated the interface and code to take in a string instead of `WeakETag`.

Reply:
Ah, ok thanks, makes sense